src/pkg/services/m365/api/events.go

```diff
@@ -208,63 +219,223 @@ func (c Events) GetItem(
+	ctx context.Context,
+	event models.Eventable,
+	immutableIDs bool,
+	userID string,
+	itemID string,
+) ([]models.Attachmentable, error) {
+	if !ptr.Val(event.GetHasAttachments()) && !HasAttachments(event.GetBody()) {
+		return nil, nil
+	}
+
+	config := &users.ItemEventsItemAttachmentsRequestBuilderGetRequestConfiguration{
+		QueryParameters: &users.ItemEventsItemAttachmentsRequestBuilderGetQueryParameters{
+			Expand: []string{"microsoft.graph.itemattachment/item"},
+		},
+		Headers: newPreferHeaders(preferPageSize(maxNonDeltaPageSize), preferImmutableIDs(immutableIDs)),
+	}
+
+	attached, err := c.LargeItem.
 		Client().
 		Users().
 		ByUserId(userID).
 		Events().
 		ByEventId(itemID).
+		Attachments().
 		Get(ctx, config)
 	if err != nil {
-		return nil, nil, graph.Stack(ctx, err)
+		return nil, graph.Wrap(ctx, err, "event attachment download")
 	}
 
-	if ptr.Val(event.GetHasAttachments()) || HasAttachments(event.GetBody()) {
-		config := &users.ItemEventsItemAttachmentsRequestBuilderGetRequestConfiguration{
-			QueryParameters: &users.ItemEventsItemAttachmentsRequestBuilderGetQueryParameters{
-				Expand: []string{"microsoft.graph.itemattachment/item"},
-			},
-			Headers: newPreferHeaders(preferPageSize(maxNonDeltaPageSize), preferImmutableIDs(immutableIDs)),
-		}
+	return attached.GetValue(), nil
+}
 
-		attached, err := c.LargeItem.
-			Client().
-			Users().
-			ByUserId(userID).
-			Events().
-			ByEventId(itemID).
-			Attachments().
-			Get(ctx, config)
-		if err != nil {
-			return nil, nil, graph.Wrap(ctx, err, "event attachment download")
-		}
+func (c Events) GetItemInstances(
+	ctx context.Context,
+	userID, itemID string,
+	startDate, endDate string,
+) ([]models.Eventable, error) {
+	config := &users.ItemEventsItemInstancesRequestBuilderGetRequestConfiguration{
+		QueryParameters: &users.ItemEventsItemInstancesRequestBuilderGetQueryParameters{
+			Select:        []string{"id"},
+			StartDateTime: ptr.To(startDate),
+			EndDateTime:   ptr.To(endDate),
+		},
+	}
 
-		event.SetAttachments(attached.GetValue())
+	events, err := c.Stable.
+		Client().
+		Users().
+		ByUserId(userID).
+		Events().
+		ByEventId(itemID).
+		Instances().
+		Get(ctx, config)
+	if err != nil {
+		return nil, graph.Stack(ctx, err)
 	}
 
-	return event, EventInfo(event), nil
+	return events.GetValue(), nil
 }
 
 func (c Events) PostItem(
 	ctx context.Context,
 	userID, containerID string,
 	body models.Eventable,
 ) (models.Eventable, error) {
-	itm, err := c.Stable.
-		Client().
-		Users().
-		ByUserId(userID).
-		Calendars().
-		ByCalendarId(containerID).
-		Events().
-		Post(ctx, body, nil)
+	rawURL := fmt.Sprintf(eventPostBetaURLTemplate, userID, containerID)
+	builder := users.NewItemCalendarsItemEventsRequestBuilder(rawURL, c.Stable.Adapter())
+
+	itm, err := builder.Post(ctx, body, nil)
 	if err != nil {
 		return nil, graph.Wrap(ctx, err, "creating calendar event")
 	}
 
 	return itm, nil
 }
 
+func (c Events) UpdateItem(
```

Comment:
Nit: since update can refer to either Put or Patch, and the difference might be important.
```suggestion
func (c Events) PatchItem(
```